### PR TITLE
BD-2013: New Update catalog items and Update catalog item endpoints

### PR DESCRIPTION
**Pull request by lydia-xie**
# Pull Request/Issue Resolution

Closes #**BD-2013**

#### Is this change associated with a Braze feature/product release?
- [ ] Yes (**Insert Feature Release Date Here**)
- [ ] No

---

### CODE REVIEW

**Comment by raghavnarain3** [comment]
@lydia-xie one of these should be async and the other should be sync it looks like both the files are in the async directory

**Comment by lydia-xie** [comment]
@raghavnarain3 - can you clarify which endpoints should be asynchronous and synchronous? the gdoc I was provided lists them both as asynchronous - https://docs.google.com/document/d/1YDjBPWQ6wjXkJvGNJccb3F8iWLWJWNvaAgVSL8CwAcE/edit

**Comment by lydia-xie** [comment]
@raghavnarain3 - updated the PR - there's an async and sync endpoint now

**Comment by raghavnarain3** [comment]
We show a helpful errors table in our other endpoints. I think we should add it here. Below is the errors for each endpoint all of them should be referenced in a different endpoint page:

Below are the error names for our async PUT
```
              # errors:
              #   400 Errors:
              #     - REQUEST_INCLUDES_TOO_MANY_ITEMS
              #     - IDS_NOT_UNIQUE
              #     - IDS_TOO_LARGE
              #     - IDS_NOT_STRINGS
              #     - INVALID_IDS
              #     - ITEMS_MISSING_IDS
              #     - ITEMS_TOO_LARGE
              #     - INVALID_FIELDS
              #     - UNABLE_TO_COERCE_VALUE
              #     - INVALID_KEYS_IN_VALUE_OBJECT
              #     - TOO_DEEP_NESTING_IN_VALUE_OBJECT
              #     - ITEM_ARRAY_INVALID
              #   404 Errors:
              #     - CATALOG_NOT_FOUND
```      

And below are the ones for a sync PUT
```
# errors:
                #   400 Errors:
                #     - REQUEST_INCLUDES_TOO_MANY_ITEMS
                #     - ID_IN_BODY
                #     - IDS_TOO_LARGE
                #     - INVALID_IDS
                #     - ITEMS_TOO_LARGE
                #     - INVALID_FIELDS
                #     - UNABLE_TO_COERCE_VALUE
                #     - INVALID_KEYS_IN_VALUE_OBJECT
                #     - TOO_DEEP_NESTING_IN_VALUE_OBJECT
                #     - ALREADY_REACHED_CATALOG_ITEM_LIMIT
                #     - ALREADY_REACHED_COMPANY_ITEM_LIMIT
                #     - ITEM_ALREADY_EXISTS
                #     - FILTERED_SET_FIELD_TOO_LONG
                #     - ARBITRARY_ERROR
                #     - ITEM_ARRAY_INVALID
                #   404 Errors:
                #     - CATALOG_NOT_FOUND
```

File: _docs/_api/endpoints/catalogs/catalog_items/asynchronous/put_update_catalog_items.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+  ]
+}'
+```
+
+## Response
+
+There are three status code responses for this endpoint: `202`, `400`, and `404`.
+
+### Example success response
```
Comment (raghavnarain3) [inline _docs/_api/endpoints/catalogs/catalog_items/asynchronous/put_update_catalog_items.md:75]: these are missing example failures as well -- similar to the other endpoints

**Review by bre-fitzgerald — APPROVED** [review]

**Pull request merged**